### PR TITLE
Breakout pid lists

### PR DESCRIPTION
**Pull request by mejackreed**
d14098b enables Resource Index to be used for development and test ... Useful for testing our indexing approach

8721a70 Moves pid gathering tasks to its own class. specs mainly test memoization functionality of collection methods

### CODE REVIEW

File: lib/pid_gatherer.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        set_pids)
+    end
+
+    private
+
+    ##
+    # Query the Dor SearchService for a specific model
+    def pids_for_model_type(model_type)
+      Dor::SearchService.risearch 'select $object from <#ri> where $object <f' \
+        "edora-model:hasModel> #{model_type}"
```
Comment (atz) [inline lib/pid_gatherer.rb]: This is pretty fugly, truncating in the middle of a word.  Go long-line instead, or at least line break before the `<f`, imho.

Comment (mejackreed) [inline lib/pid_gatherer.rb]: Agreed... fixed :clap: thanks!

**Comment by jmartin-sul** [comment]
ha, just finished reading through this.  only had one minor suggestion, which was to rename `prioritize_pid_lists` to something indicating that it also does queuing (e.g. `prioritize_pid_lists_and_queue_for_reindexing` or `prioritize_and_queue_pid_lists`, or whatever).  but otherwise, definite :+1: from me, and i can make that change on `profiling_and_indexing_improvements`, if you guys find that suggestion agreeable.

**Comment by mejackreed** [comment]
Uggg.. naming is sooo hard. :+1:

**Comment by jmartin-sul** [comment]
indeed it is

**Pull request merged**